src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -1129,60 +1112,50 @@ private void removeEvidence(final int sampleIndex, final Collection<EVIDENCE> ev
-
-                int nextIndexToRemove = evidencesToRemove[0];
-                if (nextIndexToRemove < 0) {
-                    throw new IllegalStateException("invalid input index array as it contains negatives");
-                }
-                evidenceIndex.remove(evidences.get(nextIndexToRemove));
-                for (int etrIndex = 1, to = nextIndexToRemove, from = to + 1; to < newEvidenceCount; etrIndex++, from++) {
-                    if (etrIndex < evidencesToRemove.length) {
-                        nextIndexToRemove = evidencesToRemove[etrIndex];
-                        if (nextIndexToRemove < from) {
-                            throw new IllegalStateException("invalid input index array contains indexes out of order");
-                        } else if (nextIndexToRemove >= oldEvidenceCount) {
-                            throw new IllegalStateException("invalid input index array contains indexes out of order");
-                        }
-                        evidenceIndex.remove(evidences.get(nextIndexToRemove));
-                    } else {
-                        nextIndexToRemove = oldEvidenceCount;
-                    }
-                    for (; from < nextIndexToRemove; from++) {
-                        final EVIDENCE evidence = evidences.get(from);
-                        evidences.set(to, evidence);
-                        evidenceIndex.put(evidence, to++);
-                    }
-                }
-                Utils.truncate(evidences, newEvidenceCount);
-                // now we do the likelihood values, we save ourselves all the checks:
-                for (final double[] alleleValues : values) {
-                    for (int etrIndex = 1, to = evidencesToRemove[0], from = to + 1; to < newEvidenceCount; from++, etrIndex++) {
-                        nextIndexToRemove = etrIndex < evidencesToRemove.length ? evidencesToRemove[etrIndex] : oldEvidenceCount;
-                        for (; from < nextIndexToRemove; from++) {
-                            alleleValues[to++] = alleleValues[from];
-                        }
-                    }
+                newEvidence.add(oldEvidence.get(n));
+            }
+        }
+        Utils.validate(newEvidence.size() == newEvidenceCount, "Indices to remove contained duplicates, was not ordered, or contained out-of-range indices.");
+        evidenceBySampleIndex.set(sampleIndex, newEvidence);
+        numberOfEvidences[sampleIndex] = newEvidenceCount;
+
+        invalidateEvidenceToIndexCache(sampleIndex);
+
+        // update the likelihoods arrays in place
+        for (final double[] alleleValues : valuesBySampleIndex[sampleIndex]) {
+            for (int n = 0, numRemoved = 0; n < oldEvidenceCount; n++) {
+                if (numRemoved < numToRemove && n == evidencesToRemove[numRemoved]) {
+                    numRemoved++;
+                } else {
+                    alleleValues[n - numRemoved] = alleleValues[n];
```

Comment:
why we have two for loops that seem to do pretty much the same. Can we simply move this line of code to the ```for``` loop above?... I mean:

```
for (...) {
   if (n == nextToRemove) {
     ...
   } else {
     ...
     for (final double[] alleleValues : valuesBySampleIndex[sampleIndex]) {
          alleleValues[n - numRemoved] = alleleValues[n];
     }
   }
```

Otherwise if you want to keep it this way, you could start the loop on the first removed-index:

```
   for (int n = evidencesToRemove[0] + 1, ...; n < oldEvidenceCount; n++) {
       ...
   }
```

Reply:
Your way is better.  Done.